src/main/java/org/zalando/nakadi/controller/SubscriptionController.java

```diff
@@ -143,4 +148,35 @@ private Subscription getExistingSubscription(final SubscriptionBase subscription
                 subscriptionBase.getConsumerGroup());
     }
 
+    private final class EventTypesHolder {
```

Comment:
I think this solution is pretty straight forward (though the class is cumbersome) and It will be more confusing to have a method which subtracting found event types from subscription event names. Also in that case we will have several loops but we have exception for that when calling findByName.